### ISSUE TITLE
[no-unused-expressions] directive detection does not work on modules

### ISSUE THREAD

**Issue by HolgerJeromin** [typescript-eslint/typescript-eslint#1283]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-unused-expressions": "error"
  }
}
```

```TS
module MyModule {
    'use strict';
}
```

**Expected Result**
Should not complain

**Actual Result**
````
demo.ts(1,5): error @typescript-eslint/no-unused-expressions : 
Expected an assignment or function call and instead saw an expression.
````
Same problem was happening with `no-unused-expressions` from `eslint` (but i upgraded to ts-eslint one).

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.8.0` |
| `@typescript-eslint/parser`        | `2.8.0` |
| `TypeScript`                       | `3.7.2` |
| `ESLint`                           | `6.6.0` |
